frontend/src/app/exercises-listing/exercises-listing.component.html

```diff
@@ -1,8 +1,39 @@
 <div class="exercises-page">
-    <div class="acrobatic-exercises" fxLayout="row wrap" fxLayoutAlign="center" fxLayoutGap="15px">
-        <div *ngFor="let exercise of exercises" class="single-exercise clickable" [routerLink]="['/exercises', exercise.id]">
-            <h2>{{exercise.name}}</h2>
-            <p>{{exercise.description.length > 300 ? exercise.description.substr(0,300) + '...' : exercise.description}}</p>
+    <div fxLayout="column" fxLayout.gt-sm="row">
+        <div class="padding" fxFlex.gt-sm="350px" fxFlexOrder.lt-md="3">
+            <mat-accordion>
+                <mat-expansion-panel>
+                    <mat-expansion-panel-header>
+                        <mat-panel-title>
+                            <b>{{'COMMON.FILTERING' | translate}}</b>
+                        </mat-panel-title>
+                    </mat-expansion-panel-header>
+                    <hr>
+                    <p>{{'EXERCISES.EXERCISES_DIFFICULTY' | translate}}</p>
+                    <mat-form-field class="example-form-field" appearance="fill">
+                        <mat-label>Min</mat-label>
+                        <input matInput class="minDifficulty-input" type="number" min="1" max="10" [ngModel]="minDifficulty" (ngModelChange)="updateMinDifficulty($event)">
+                    </mat-form-field>
+                    <mat-form-field class="example-form-field" appearance="fill">
+                        <mat-label>Max</mat-label>
+                        <input matInput class="maxDifficulty-input" type="number" min="1" max="10" [ngModel]="maxDifficulty" (ngModelChange)="updateMaxDifficulty($event)">
+                    </mat-form-field>
+                    <hr>
+                    <p>{{'COMMON.SEARCH' | translate}}</p>
+                    <mat-form-field class="example-form-field" appearance="fill">
+                        <mat-label></mat-label>
+                        <input class="query-input" matInput type="text" [ngModel]="query" (ngModelChange)="updateQuery($event)">
+                    </mat-form-field>
+                </mat-expansion-panel>
+            </mat-accordion>
         </div>
+        <div class="exercises-list" fxLayout="row wrap" fxLayoutAlign="center" fxLayoutGap="15px" fxFlex="grow">
+            <div *ngFor="let exercise of filteredExercises" class="single-exercise clickable" [routerLink]="['/exercises', exercise.id]">
+                <h2>{{exercise.name}}</h2>
+                <strong>{{'EXERCISES.DIFFICULTY' | translate}} - {{exercise.difficulty}}</strong>
+                <p>{{exercise.description.length > 300 ? exercise.description.substr(0,300) + '...' : exercise.description}}</p>
```

Comment:
**suggestion:** as a next step, can you please try to make a pipe out of this? https://angular.io/guide/pipes